### PR TITLE
Add Developer Tools section in docs

### PR DESCRIPTION
**Pull request by brbrr**
#### Changes proposed in this Pull Request:

* Add Developer tools section in Docs

#### Testing instructions:

* read the docs
* check for typos

#### Proposed changelog entry for your changes:

None!

### CODE REVIEW

File: docs/development-environment.md [hunk 1]
```diff
@@ -206,9 +206,48 @@ _If you haven't done it yet, you may need to run `yarn` before `yarn lint` for i
...
+
+You should do all Jetpack development with `define( 'WP_DEBUG', true );` in your `wp-config.php`, making sure that you’re not generating any Notices or other PHP issues in your error_log.
+
+### WP-CLI
+
+Jetpack CLI is a command line interface for Jetpack, extending off of wp-cli for WordPress. You can easily modify your installation of Jetpack with a just a few simple commands. All you need is SSH access and a basic understanding of command line tools.
+
+Usage:
+
+* wp jetpack status [<full>]
```
Comment (jeherve) [inline docs/development-environment.md]: Maybe this should be in a code block?

File: docs/development-environment.md [hunk 2]
```diff
@@ -206,9 +206,48 @@ _If you haven't done it yet, you may need to run `yarn` before `yarn lint` for i
...
+* wp jetpack module <list|activate|deactivate|toggle> [<module_name>]
+* wp jetpack options <list|get|delete|update> [<option_name>] [<option_value>]
+* wp jetpack protect <whitelist> [<ip|ip_low-ip_high|list|clear>]
+* wp jetpack reset <modules|options>
+* wp jetpack disconnect <blog|user> [<user_identifier>]
+* wp jetpack status
+* wp jetpack status [<full>]
+
+More info can be found in [Jetpack support documentation](https://jetpack.com/support/jetpack-cli/).
```
Comment (jeherve) [inline docs/development-environment.md]: Maybe in "our support documentation"?

File: docs/development-environment.md [hunk 3]
```diff
@@ -206,9 +206,48 @@ _If you haven't done it yet, you may need to run `yarn` before `yarn lint` for i
...
+
+### JETPACK_DEV_DEBUG
+
+`JETPACK_DEV_DEBUG` constant can be used to enable development mode in Jetpack. With Development Mode, features that do not require a connection to WordPress.com servers can be activated on a localhost WordPress installation for testing.
```
Comment (jeherve) [inline docs/development-environment.md]: Maybe just `local` instead of `localhost`?

File: docs/development-environment.md [hunk 4]
```diff
@@ -206,9 +206,48 @@ _If you haven't done it yet, you may need to run `yarn` before `yarn lint` for i
...
+
+Development mode automatically gets enabled if you don’t have a period in your site’s hostname, i.e. localhost. If you use a different URL, such as mycooltestsite.local or something, then you will need to define the JETPACK_DEV_DEBUG constant.
```
Comment (jeherve) [inline docs/development-environment.md]: I think you could remove `or something`

Comment (jeherve) [inline docs/development-environment.md]: I would add code fences around `JETPACK_DEV_DEBUG`

File: docs/development-environment.md [hunk 5]
```diff
@@ -206,9 +206,48 @@ _If you haven't done it yet, you may need to run `yarn` before `yarn lint` for i
...
+
+You can also enable Jetpack’s development mode through a plugin, thanks to the jetpack_development_mode filter:
+
+`add_filter( 'jetpack_development_mode', '__return_true' );`
+
+While in Development Mode, some features will not be available at all as they require WordPress.com for all functionality—Related Posts and Publicize, for example. Other features will have reduced functionality to give developers a good-faith representation of the feature. For example, Tiled Galleries requires the WordPress.com Photon CDN; however, in Development Mode, Jetpack provides a fallback so developers can have a similar experience during development and testing.
+
```
Comment (jeherve) [inline docs/development-environment.md]: Maybe you could link to the support doc here for folks who want to find out more?

File: docs/development-environment.md [hunk 6]
```diff
@@ -206,9 +206,48 @@ _If you haven't done it yet, you may need to run `yarn` before `yarn lint` for i
...
+* wp jetpack disconnect <blog|user> [<user_identifier>]
+* wp jetpack status
+* wp jetpack status [<full>]
+
+More info can be found in [Jetpack support documentation](https://jetpack.com/support/jetpack-cli/).
+
+### JETPACK_DEV_DEBUG
+
+`JETPACK_DEV_DEBUG` constant can be used to enable development mode in Jetpack. With Development Mode, features that do not require a connection to WordPress.com servers can be activated on a localhost WordPress installation for testing.
+
```
Comment (jeherve) [inline docs/development-environment.md]: In the paragraph above about `WP_DEBUG, you mentioned how to actually use it (`define( 'WP_DEBUG', true );` in your `wp-config.php`). Should we have something similar here for `JETPACK_DEV_DEBUG` and for `JETPACK__SANDBOX_DOMAIN`?

**Review by jeherve — DISMISSED** [review]
Good addition, it's going to be helpful!

I left a few notes about things you could change. 👍

File: docs/development-environment.md [hunk 7]
```diff
@@ -206,9 +206,48 @@ _If you haven't done it yet, you may need to run `yarn` before `yarn lint` for i
...
+You can also enable Jetpack’s development mode through a plugin, thanks to the jetpack_development_mode filter:
+
+`add_filter( 'jetpack_development_mode', '__return_true' );`
+
+While in Development Mode, some features will not be available at all as they require WordPress.com for all functionality—Related Posts and Publicize, for example. Other features will have reduced functionality to give developers a good-faith representation of the feature. For example, Tiled Galleries requires the WordPress.com Photon CDN; however, in Development Mode, Jetpack provides a fallback so developers can have a similar experience during development and testing. Find out more in [our support documentation](https://jetpack.com/support/jetpack-for-developers/).
+
+### JETPACK__SANDBOX_DOMAIN
+
+External contributors shouldn't use this constant.
```
Comment (mdawaffe) [inline docs/development-environment.md]: Maybe the following instead?

> External contributors do not need this constant.

**Review by jeherve — APPROVED** [review]
This looks good to me. 🚢

**Pull request merged**